### PR TITLE
task 1 complete and ready for code review

### PR DESCRIPTION
**Pull request by nesclark**

### CODE REVIEW

File: postdata.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+if (!filter_var($image_sanitized, FILTER_VALIDATE_URL) === false) {
+    echo 'valid image url' . '<br>';
+} else {
+    echo 'invalid image url' . '<br>';
+}
+
+
+
+//echo '<pre>';
```
Comment (Steph-ski) [inline postdata.php]: if this is no longer needed, remove rather than leave as comment

File: addplant.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        </li>
+    </ul>
+</nav>
+<header>
+    <div class="input-container">
+        <h3>Add Plant</h3>
+        <form action="postdata.php" method="post">
+            <label for="colloquial_name">Colloquial name: </label>
+            <input type="text" name="colloquial_name" />
+            <label for="latin_name">Latin name: </label>
```
Comment (Steph-ski) [inline addplant.php:33]: HTML validator picked up: The value of the for attribute of the label element must be the ID of a non-hidden form control.
 Looks like 'for' needs changing to 'id' within 'input'

File: addplant.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            <input type="text" name="latin_name" />
+            <label for="size_cm">Size (cm): </label>
```
Comment (Steph-ski) [inline addplant.php:35]: HTML validator picked up: The value of the for attribute of the label element must be the ID of a non-hidden form control.
 Looks like 'for' needs changing to 'id' within 'input'

File: addplant.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            <input type="number" name="size_cm" />
+            <label for="image_url">Image URL: </label>
```
Comment (Steph-ski) [inline addplant.php:37]: HTML validator picked up: The value of the for attribute of the label element must be the ID of a non-hidden form control.
 Looks like 'for' needs changing to 'id' within 'input'

File: postdata.php [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+$size_input = $_POST['size_cm'];
+$image_input = $_POST['image_url'];
+
+
+$colloquial_name_sanitized = filter_var($colloquial_name_input, FILTER_SANITIZE_STRING);
+$latin_name_sanitized = filter_var($latin_name_input, FILTER_SANITIZE_STRING);
+$size_sanitized = filter_var($size_input, FILTER_SANITIZE_NUMBER_INT);
+$image_sanitized = filter_var($image_input, FILTER_SANITIZE_URL);
+
+
```
Comment (Steph-ski) [inline postdata.php]: can this be functionalised and added to the functions.php doc? Could use && to add all conditions together and output 'valid' or 'invalid'.

If functionalised and moved to functions.php doc, include dock block.

File: postdata.php [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    echo 'invalid plant name' . '<br>';
+}
+
+if (preg_match('/^[a-zA-Z\s]+$/', $latin_name_sanitized)) {
+    echo 'valid plant name' . '<br>';
+} else {
+    echo 'invalid plant name' . '<br>';
+}
+
+if (!filter_var($size_sanitized, FILTER_VALIDATE_INT) === false) {
```
Comment (Steph-ski) [inline postdata.php]: this looks like 2 negatives - could remove ! and ===false , which would leave it as true?! I may be wrong though!

File: postdata.php [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    echo 'valid size' . '<br>';
+} else {
+    echo 'invalid size' . '<br>';
+}
+
+if (!filter_var($image_sanitized, FILTER_VALIDATE_URL) === false) {
```
Comment (Steph-ski) [inline postdata.php]: this looks like 2 negatives - could remove ! and ===false , which would leave it as true?! I may be wrong though!

**Review by Steph-ski — COMMENTED** [review]
Small amends picked up by HTML validator on the form code. 

Suggestion to functionalising sanitising and validating inputs. 

I tried two URL for images, both were reported back as invalid.

File: postdata.php (hunk 1 above)
Comment (nesclark) [reply postdata.php]: done

File: addplant.php (hunk 2 above)
Comment (nesclark) [reply addplant.php:33]: have added ids for all inputs

File: postdata.php (hunk 6 above)
Comment (nesclark) [reply postdata.php]: fixed this

File: postdata.php (hunk 7 above)
Comment (nesclark) [reply postdata.php]: fixed this

File: postdata.php (hunk 5 above)
Comment (nesclark) [reply postdata.php]: functionalised and unit tested

**Review by Steph-ski — APPROVED** [review]
All comments address. 

New function unit tested - all passed. 

Tested form input.

**Pull request merged**